src/binary/paths.ts

```diff
@@ -5,68 +5,103 @@ import {
   BINARY_UPDATE_URL,
   BINARY_UPDATE_VERSION_FILE_URL,
 } from "../globals/consts";
+import { ONPREM } from "../onPrem";
 
-let binaryRootPath: string | undefined;
+let BINARY_ROOT_PATH: string | undefined;
 const ARCHITECTURE = getArch();
 const SUFFIX = getSuffix();
 const BUNDLE_SUFFIX = getBundleSuffix();
 
 export async function setBinaryRootPath(
   extensionContext: vscode.ExtensionContext
 ): Promise<void> {
-  binaryRootPath =
+  if (ONPREM) {
+    const base = `binaries/${getArch()}-${getPlatform()}`;
```

Comment:
why ? all this is already implemented, you can use this https://github.com/codota/tabnine-vscode/blob/dc35e8a59c7ecd0e074c893fc0a58f5bfb3e1c20/src/binary/paths.ts#L29

Reply:
there is no version . it is bundled flat

Reply:
ok , but all the path logic is already implemented there, you can extract it or something